### PR TITLE
Make documentation settings script-configurable from the command line

### PR DESCRIPTION
**Pull request by wrouesnel**
The current Github Pages publish scripts depend on aspects of uniform, modified here. This PR changes the way we set these constants so that rather then being hardcoded, they're specified as Java defines so they can be injected by the sbt-ci-build-doc script.

Currently I set defaults for these so that regular `sbt` builds will work without needing to be explicit. But, it would also be possible to simply default them to the original values. As this affects everyone using uniform, some discussion on the best approach is needed.

### CODE REVIEW

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/standard/StandardProjectPlugin.scala [hunk 1]
```diff
@@ -64,14 +64,14 @@ object StandardProjectPlugin extends Plugin {
       unidocSettings ++ site.settings ++ Seq(
         site.addMappingsToSiteDir(mappings in (ScalaUnidoc, packageDoc), "latest/api"),
         includeFilter in makeSite := "*.html" | "*.css" | "*.png" | "*.jpg" | "*.gif" | "*.js" | "*.swf" | "*.md" | "*.yml",
-        apiURL <<= baseDirectory(base => Some(url(s"https://commbank.github.io/${base.getName}/latest/api"))),
+        apiURL <<= baseDirectory(base => Some(url(System.getProperty("cba.doc.rooturl", "http://127.0.0.1/")))),
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/standard/StandardProjectPlugin.scala]: I liked your settings approach better. Also this approach is backwards incompatible.

**Comment by wrouesnel** [comment]
We now use the backwards compatible approach. Test builds on my local machine work fine with both injected and default settings.

**Comment by wrouesnel** [comment]
@stephanh Force pushed and updated.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/standard/StandardProjectPlugin.scala [hunk 2]
```diff
@@ -41,6 +42,9 @@ object StandardProjectPlugin extends Plugin {
   }
 
   object uniform {
+    lazy val docRootUrl = SettingKey[String]("doc-root-url")
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/standard/StandardProjectPlugin.scala]: Setting keys take a second parameter which is a description of what the key does. Can you please add that.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/standard/StandardProjectPlugin.scala [hunk 3]
```diff
@@ -62,14 +66,16 @@ object StandardProjectPlugin extends Plugin {
     /** Settings to create content for github pages. Should only be use by the root project.*/
     def ghsettings: Seq[sbt.Setting[_]] =
       unidocSettings ++ site.settings ++ Seq(
+        docRootUrl := "https://commbank.github.io",
+        docSourceUrl := "https://github.com/CommBank",
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/standard/StandardProjectPlugin.scala]: Line up the `:=` :smile:

**Comment by stephanh** [comment]
:+1: After the minor comments.

**Pull request merged**